__tests__/app.test.js

```diff
@@ -11,4 +12,86 @@ describe('backend-top-secrets routes', () => {
   afterAll(() => {
     pool.end();
   });
+
+  it('registers a user', async () => {
+    const res = await request(app)
+      .post('/api/v1/users/register')
+      .send({ email: 'test@gmail.com', password: 'password' });
+
+    expect(res.body).toEqual({
+      id: expect.any(String),
+      email: 'test@gmail.com',
+    });
+  });
+
+  it('signs in an existing user', async () => {
+    const user = await UserService.create({
+      email: 'test@gmail.com',
+      password: 'password',
+    });
+
+    const res = await request(app)
+      .post('/api/v1/users/session')
+      .send({ email: 'test@gmail.com', password: 'password' });
+
+    expect(res.body).toEqual({
+      message: 'Signed in successfully!',
+      user,
+    });
+  });
+
+  it('allows a signed in user to view secrets', async () => {
+    const agent = request.agent(app);
+
+    await UserService.create({
+      email: 'test2@gmail.com',
+      password: 'password',
+    });
+
+    let res = await agent.get('/api/v1/secrets');
+    expect(res.status).toEqual(401);
+
+    await agent
+      .post('/api/v1/users/session')
+      .send({ email: 'test2@gmail.com', password: 'password' });
+    res = await agent.get('/api/v1/secrets');
+    expect(res.status).toEqual(200);
+  });
+
+  it('allows a signed in user to create secrets', async () => {
+    const agent = request.agent(app);
+
+    const secret = {
+      title: 'This is my secret',
+      description: 'One time I went to the store',
+      createdAt: expect.any(String),
+    };
+
+    await UserService.create({
+      email: 'test3@gmail.com',
+      password: 'password',
+    });
+    await agent
+      .post('/api/v1/users/session')
+      .send({ email: 'test3@gmail.com', password: 'password' });
+
+    const res = await request(app).post('/api/v1/secrets').send(secret);
```

Comment:
You'd need to use your agent in this post to make use of the cookie stored from the login just before. Posting to request(app) is the equivalent of an unauthenticated user posting, which means it should not be successful.